### PR TITLE
added lock wait timeout counter and appropriately incremented

### PR DESCRIPTION
**Pull request by armishra**
Changes:
* modified test and added expected result
* changed stats from incrementing to setting

### CODE REVIEW

File: mysql-test/suite/rocksdb/t/lock_wait_timeout_stats.test [hunk 1]
```diff
@@ -0,0 +1,27 @@
+--source include/have_rocksdb.inc
+
+#
+# MyRocks-specific tests for locking
+#
+--source include/have_debug.inc
+
+--enable_connect_log
+--disable_warnings
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/lock_wait_timeout_stats.test]: L9 to L11: obsolete idiom

File: mysql-test/suite/rocksdb/t/lock_wait_timeout_stats.test [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+drop table if exists t;
+--enable_warnings
+create table t (a int not null primary key) engine=rocksdb;
+
+begin;
+insert into t values (0);
+
+--connect (con1,localhost,root,,)
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/lock_wait_timeout_stats.test]: Consider adding --source include/count_sessions.inc above connect and...

File: mysql-test/suite/rocksdb/t/lock_wait_timeout_stats.test [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+--connection con1
+set @@rocksdb_lock_wait_timeout=1;
+begin;
+--error ER_LOCK_WAIT_TIMEOUT
+insert into t values(0);
+select INNODB_ROW_LOCK_WAIT_TIMEOUTS from information_schema.table_statistics where table_engine="ROCKSDB" and table_name="t";
+select INNODB_ROW_LOCK_WAIT_TIMEOUTS from information_schema.table_statistics where table_engine="ROCKSDB" and table_name="t";
+--disconnect con1
+--connection default
+drop table t;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/lock_wait_timeout_stats.test:25]: ... --source include/wait_until_count_sessions.inc somewhere after the disconnect for the testcase stability

File: sql/table_stats.cc [hunk 4]
```diff
@@ -441,7 +441,7 @@ void fill_table_stats_cb(const char *db,
   copy_comp_stats_with_races(&stats->comp_stats, comp_stats);
 
   stats->n_lock_wait.inc(n_lock_wait);
-  stats->n_lock_wait_timeout.inc(n_lock_wait_timeout);
+  stats->n_lock_wait_timeout.set_maybe(n_lock_wait_timeout);
```
Comment (alxyang) [inline sql/table_stats.cc]: One issue I can see for this is that it may break InnoDB stats, since they also go through this code path.

**Review by alxyang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**